config/flask_config.py

```diff
@@ -2,80 +2,125 @@
-        try:
-            with open(CLIENT_SECRETS_PATH, 'r') as f:
-                _secrets_data = json.loads(f.read())['web']
-                GOOGLE_CLIENT_ID = _secrets_data['client_id']
-                if not _secrets_data.get('client_secret', None):
-                    print ('Google Auth config file, %s,'
-                           ' missing client secret', CLIENT_SECRETS_PATH)
-                    exit(1)
-
-        except IOError:
-            print ("The Google client_secrets file was not found at"
-                   "'{}', check that it exists.".format(CLIENT_SECRETS_PATH))
-            exit(1)
-
-    # Cross-site request forgery settings
-    CSRF_ENABLED = True if environ['CSRF_ENABLED'] == 'TRUE' else False
-    CSRF_SESSION_KEY = environ['CSRF_SESSION_KEY']
-
-    # Google Calendar credentials
-    PRIVATE_CALENDAR_ID = environ['PRIVATE_CALENDAR_ID']
-    PUBLIC_CALENDAR_ID =  environ['PUBLIC_CALENDAR_ID']
-
-    # Mongo configs
-    MONGODB_SETTINGS = {'DB': environ.get('MONGO_DATABASE', 'eventum')}
-
-    # Logging settings
-    LOG_FILE_MAX_SIZE = environ.get("LOG_FILE_MAX_SIZE")
-    APP_LOG_NAME = environ.get("APP_LOG_NAME")
-    WERKZEUG_LOG_NAME = environ.get("WERKZEUG_LOG_NAME")
-
-except KeyError:
-    """ Throw an error if a setting is missing """
-    print ("Some of your settings aren't in the environment."
-    "You probably need to run:\n\n\tsource config/<your settings file>")
-    exit(1)
+# dictionary the flask app configures itself from
+config = {
+    'HOST': '0.0.0.0',
+    'PORT': None,
+    'SECRET_KEY': None,
+    'INSTALLED_APP_CREDENTIALS_PATH': None,
+    'CREDENTIALS_PATH': None,
+    'GOOGLE_AUTH_ENABLED': None,
+    'CLIENT_SECRETS_PATH': None,
+    'CSRF_ENABLED': None,
+    'CSRF_SESSION_KEY': None,
+    'DEBUG': False,
+    'PRIVATE_CALENDAR_ID': None,
+    'PUBLIC_CALENDAR_ID': None,
+    'MONGO_DATABASE': None,
+    'LOG_FILE_MAX_SIZE': None,
+    'APP_LOG_NAME': None,
+    'WERKZEUG_LOG_NAME': None
+}
 
+# consul_configurations contains equivalent keys that will be used to extract
+# configuration values from Consul.
+consul_configurations = [  # consul key --> config key
+    ('flask_port', 'PORT'),
+    ('flask_debug', 'DEBUG'),
+    ('secret_key', 'SECRET_KEY'),
+    ('installed_app_client_secret_path', 'INSTALLED_APP_CLIENT_SECRET_PATH'),
+    ('installed_app_credentials_path', 'INSTALLED_APP_CREDENTIALS_PATH'),
+    ('google_auth_enabled', 'GOOGLE_AUTH_ENABLED'),
+    ('client_secrets_path', 'CLIENT_SECRETS_PATH'),
+    ('csrf_enabled', 'CSRF_ENABLED'),
+    ('csrf_session_key', 'CSRF_SESSION_KEY'),
+    ('private_calendar_id', 'PRIVATE_CALENDAR_ID'),
+    ('public_calendar_id', 'PUBLIC_CALENDAR_ID'),
+    ('mongo_database', 'MONGO_DATABASE'),
+    ('log_file_max_size', 'LOG_FILE_MAX_SIZE'),
+    ('app_log_name', 'APP_LOG_NAME'),
+    ('werkzeug_log_name', 'WERKZEUG_LOG_NAME')
+]
+
+# config variables that must be set for the app to work
+critical_config_vars = [
```

Comment:
We used `environ.get` only if there was a default that we wanted to preserve.  The equivalent thing to do here would be to put default values in the dictionary up top and then populate all the values (even if there are defaults) from consul / environ and then if any key is `None` in the dict, panic